docs/site/Transactions.md

```diff
@@ -0,0 +1,201 @@
+    super(Note, crudDs);
+  }
+}
+```
+
+#### Isolation levels
+
+When you call `beginTransaction()`, you can optionally specify a transaction
+isolation level. LoopBack transactions support the following isolation levels:
+
+- `Transaction.READ_UNCOMMITTED`
+- `Transaction.READ_COMMITTED` (default)
+- `Transaction.REPEATABLE_READ`
+- `Transaction.SERIALIZABLE`
+
+If you don't specify an isolation level, the transaction uses READ_COMMITTED .
+
+{% include important.html content="
+
+**Oracle only supports READ_COMMITTED and SERIALIZABLE.**
+
+" %}
+
+For more information about database-specific isolation levels, see:
+
+- [MySQL SET TRANSACTION Syntax](https://dev.mysql.com/doc/refman/5.7/en/set-transaction.html)
+- [Oracle Isolation Levels](http://docs.oracle.com/cd/B14117_01/server.101/b10743/consist.htm#i17856)
+- [PostgreSQL Transaction Isolation](http://www.postgresql.org/docs/9.4/static/transaction-iso.html)
+- [SQL Server SET TRANSACTION ISOLATION LEVEL](https://msdn.microsoft.com/en-us/library/ms173763.aspx)
+
+### Perform operations in a transaction
+
+To perform create, retrieve, update, and delete operations in the transaction,
+add a second argument consisting of the transaction object to the standard 
+[`create()`](https://loopback.io/doc/en/lb4/apidocs.repository.defaultcrudrepository.create.html),
+[`update()`](https://loopback.io/doc/en/lb4/apidocs.repository.defaultcrudrepository.update.html),
+[`deleteAll()`](https://loopback.io/doc/en/lb4/apidocs.repository.defaultcrudrepository.deleteall.html)
+(and so on) methods. This argument is typically an `Options` object that is
+passed on to the CRUD methods.
+
+For example, again assuming a `Note` model, `repo` transactional repository, and
+transaction object `tx` created as demonstrated in
+[Start transaction](#start-transaction) section:
+
+```typescript
+const created = await repo.create({title: 'Groceries'}, {transaction: tx});
+const updated = await repo.update(
+  {title: 'Errands', id: created.id},
+  {transaction: tx},
+);
+
+// commit the transaction to persist the changes
+await tx.commit();
+```
+
+Propagating a transaction is explicit by passing the transaction object via the
+options argument for all create, retrieve, update, and delete and relation
+methods.
+
+### Commit or rollback
+
+Transactions allow you either to commit the transaction and persist the CRUD
+behaviour onto the database or rollback the changes. The two methods available
+on transaction objects are as follows:
+
+```typescript
+  /**
+   * Commit the transaction
+   */
+  commit(): Promise<void>;
+
+  /**
+   * Rollback the transaction
+   */
+  rollback(): Promise<void>;
+```
+
+## Set up timeout
+
+You can specify a timeout (in milliseconds) to begin a transaction. If a
+transaction is not finished (committed or rolled back) before the timeout, it
+will be automatically rolled back upon timeout by default. The timeout event can
+be trapped using the timeout hook.
```

Comment:
> The timeout event can be trapped using the timeout hook.

This may need more explanation - what is the "timeout hook" and how can it be used?

Reply:
Actually I used LB3 docs as a base when writing the docs. After looking at our `TransactionMixin` class in juggler and comparing it to the `Transaction` class in `loopback-connector`, we still have support for `timeout`, but won't have support for hooks on transactions for observing events before `commit`, `rollback`, or `timeout`. Thus, I'll remove this bit.